atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/RangeOfTimestampsIterator.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright 2017 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the BSD-3 License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://opensource.org/licenses/BSD-3-Clause
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.keyvalue.cassandra;
+
+import java.util.Arrays;
+import java.util.Set;
+import java.util.SortedMap;
+
+import com.google.common.collect.ImmutableSet;
+import com.google.common.collect.Iterators;
+import com.google.common.collect.Maps;
+import com.google.common.collect.PeekingIterator;
+import com.google.common.primitives.UnsignedBytes;
+import com.palantir.atlasdb.keyvalue.api.CandidateCellForSweeping;
+import com.palantir.atlasdb.keyvalue.api.CandidateCellForSweepingRequest;
+import com.palantir.atlasdb.keyvalue.api.ImmutableCandidateCellForSweepingRequest;
+import com.palantir.atlasdb.keyvalue.api.KeyValueService;
+import com.palantir.atlasdb.keyvalue.api.RangeRequest;
+import com.palantir.atlasdb.keyvalue.api.RowResult;
+import com.palantir.atlasdb.keyvalue.api.TableReference;
+import com.palantir.common.base.ClosableIterator;
+
+public class RangeOfTimestampsIterator implements ClosableIterator<RowResult<Set<Long>>> {
```

Comment:
Could definitely use a paragraph of javadoc here to explain the purpose of the class.

Reply:
Yeah, this class was kind of added as an afterthought. my original plan was to remove `getRangeOfTimestamps` since it it's not really needed now that we have `getCandidateCellsForSweeping`. However it's used for some tangential reasons here an there (in tests, and in `AbstractKeyValueService#deleteRange`), so I just deprecated it instead.

Reply:
was planning to do a follow up to see if we can just get rid of it entirely

Reply:
On the other hand if we can't get rid of this, maybe we just want to make `GetCandidateCellsForSweeping` explicitly return full row objects, and take a start and end row. It adds more complexity there but we wouldn't need this class.